Docs: TCP/IP windows

TMRh20 opened the pull request:
Adjust comments per #49 because these changes can severely impact other network services/functionality.

TMRh20 committed:
Docs: TCP/IP windows

TMRh20 committed:
Add a space

2bndy5 commented on docs/main_page.md:
````diff
@@ -70,12 +70,14 @@ In this case the master node could be configured with IP 10.1.3.1 netmask 255.25
 sudo sysctl net.ipv4.tcp_wmem="1500 1500 1500"
 sudo sysctl net.ipv4.tcp_rmem="1500 1500 1500"
 ```
-These changes are only temporary and need to be run each time before startup of the gateway. To make them permanent, edit `/etc/sysctl.conf`
+These changes are only temporary and need to be run each time after reboot of the gateway. To make them permanent, edit `/etc/sysctl.conf`
````
This wording is a bit confusing.

Before, I understood that the commands need to be run before the gateway is run. 
Now it sounds like the commands need to be run after the gateway ends and before it starts back up.

Is it supposed to refer to the entire OS reboot? Is "gateway" not just an app/server?

TMRh20 replied:
Yeah, I guess it is confusing. Its after reboot of the OS. Maybe I should just say OS?

2bndy5 replied:
Better to be explicit. I didn't know "gateway" is synonymous with the machine running this software.

TMRh20 committed:
Docs: Clarify comments

TMRh20 committed:
Update main_page.md

- Move throughput info down
- Change IP config to default RF24Ethernet example IP for the gateway

2bndy5 commented on docs/main_page.md:
````diff
@@ -91,6 +83,15 @@ sudo sysctl -w net.ipv4.ip_forward=1
 ```
 
+To maximize throughput between Raspberry Pi or other Linux devices, run the following commands:
+@note These commands can severely impact other network functionality. Run `sudo sysctl net.ipv4.tcp_wmem` and `sudo sysctl net.ipv4.tcp_rmem` to get the default settings so they can be restored.
````
ref https://github.com/nRF24/RF24Ethernet/pull/53#discussion_r2072238943
```suggestion
@warning These commands can severely impact other network functionality. Run `sudo sysctl net.ipv4.tcp_wmem` and `sudo sysctl net.ipv4.tcp_rmem` to get the default settings so they can be restored.
```

TMRh20 committed:
Update docs/main_page.md

Co-authored-by: Brendan <2bndy5@gmail.com>

TMRh20 commented:
With these final changes I'm looking towards a release crusade very soon.

What are the statuses of your PRs? Are there any you want/need to get in beforehand?

2bndy5 commented:
The preferred RF24 merge order:
1. https://github.com/nRF24/RF24/pull/1027 (tested, but won't affect lib code)
2. https://github.com/nRF24/RF24/pull/1032 (tested, just did another review)
3. https://github.com/nRF24/RF24/pull/1030 (requires rebase on pr2 to update migration guide)

Network layer(s) will also need a patch once https://github.com/nRF24/RF24/pull/1030 is merged.

TMRh20 commented:
> The preferred RF24 merge order:

Haha, so does that mean you want to get all these in before a release?

I'm not sure about RF24Network requiring changes, it mostly uses Pipe 0 for multicast and I haven't been able to test or figure out if its affected by the pipe0 bug by default.

2bndy5 commented:
Depends. If release crusade includes a patch to RF24, then just nRF24/RF24#1027
nRF24/RF24#1032 and nRF24/RF24#1030 should be a minor bump to RF24.

2bndy5 commented:
I was thinking the exposure of `ce()` should ship with the exposure of the status byte. The `stopListening()` overload is meant to be a regression fix for the pipe0 initial fix.

TMRh20 commented:
Ok, I really want to get this RF24 Dynamic Payloads bug-fix released ASAP since it causes random "apparent" radio hardware failures.

I never seem to know if your PRs are ready. If u don't mind, use the request review thing for anything you want me to review/test right away.
